grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -740,14 +741,19 @@ class StockMovementService {
     void addStockListItemsToRequisition(StockMovement stockMovement, Requisition requisition) {
         // If the user specified a stocklist then we should automatically clone it as long as there are no
         // requisition items already added to the requisition
+        RequisitionItemSortByCode sortByCode = stockMovement.stocklist?.sortByCode ?: RequisitionItemSortByCode.CATEGORY
+        def orderIndex = 0
+
         if (stockMovement.stocklist && !requisition.requisitionItems) {
-            stockMovement.stocklist.requisitionItems.each { stocklistItem ->
+            stockMovement.stocklist."${sortByCode.methodName}".each { stocklistItem ->
```

Comment:
The `sortByCode` attribute could be null, so we'll need to protect against that.

Reply:
if it's null I'm using RequisitionItemSortByCode.CATEGORY, look few lines above